js/ui/codeView.js

```diff
@@ -106,6 +106,47 @@ function _getWindowId(win) {
     return ('window:%d').format(win.get_stable_sequence());
 };
 
+const _ensureHackDataFile = (function () {
+    let keyfile = new GLib.KeyFile();
+    let initialized = false;
+
+    const flatpakPath = '/var/lib/flatpak';
+    const fileRelPath = 'share/hack-components/hack-data.ini';
+    const filePath = `${flatpakPath}/app/com.endlessm.HackComponents/current/active/files/${fileRelPath}`;
+
+    return function() {
+        if (initialized)
+            return keyfile;
+
+        try {
+            keyfile.load_from_file(filePath, GLib.KeyFileFlags.NONE);
+        } catch (err) {
+            logError(err, 'Error reading hack data file');
+            keyfile = null;
+        }
+
+        initialized = true;
+        return keyfile;
+    }
+}());
+
+function _isBlacklistedApp(desktop_id) {
+    const keyfile = _ensureHackDataFile();
+    if (keyfile === null)
+        return false;
+
+    const app_id = desktop_id.slice(0, -8);  // remove ".desktop"
+    let blacklist;
+    try {
+        [blacklist] = keyfile.get_string_list('flip-to-hack', 'blacklist');
+    } catch (err) {
+        logError(err, 'No blacklist in hack data file');
```

Comment:
I don't think we should log an error if the group name or key name are missing, since it could just be that we removed the blacklist but have other uses for this file. (Or alternatively, look up this information from a file that explicitly only has the blacklist in it, instead of a more general "hack data" keyfile.)